e5e.go

```diff
@@ -1,261 +1,140 @@
-package e5e // import "go.anx.io/e5e"
-
-import (
-	"bufio"
-	"encoding/json"
-	"fmt"
-	"os"
-	"reflect"
-	"runtime"
-	"strings"
-)
-
-// Init to enforce handling of the metadata command used by e5e to determine the supported feature set.
-func init() {
-	type metadata struct {
-		LibraryVersion string   `json:"library_version"`
-		Runtime        string   `json:"runtime"`
-		RuntimeVersion string   `json:"runtime_version"`
-		Features       []string `json:"features"`
-	}
-
-	if len(os.Args) == 2 && os.Args[1] == "metadata" {
-		metadataInstance := metadata{
-			LibraryVersion: LibraryVersion,
-			Runtime:        "Go",
-			RuntimeVersion: runtime.Version(),
-			Features:       []string{"keepalive"},
-		}
-		metadataBytes, err := json.Marshal(metadataInstance)
-		if err != nil {
-			panic(fmt.Errorf("go-e5e: metadata generation failed: %w", err))
-		}
-		_, _ = os.Stdout.Write(metadataBytes)
-		os.Exit(0)
-	}
-}
```

Comment:
Moving the metadata handling to our global `Start` function could have unwanted side effects. The customer could choose to run some heavy calculations or try to access the network before our library gets control. This is not intended for metadata fetching.
During our deployment the metadata fetcher executes the binary within a very restricted environment (CPU resources, memory, no network access and a short execution timeout). Metadata fetching should still work even though the actual execution relies on a real function execution environment.